src/main/java/decodes/sql/PlatformListIO.java

```diff
@@ -421,129 +417,102 @@ private TransportMedium rs2tm(ResultSet rs, Platform platform)
 	public void readPlatform(Platform p)
 		throws SQLException, DatabaseException
 	{
-		try (Statement stmt = createStatement();)
-		{
-			String q = "SELECT " + getColTpl() + " FROM Platform WHERE ID = " + p.getId();
-			debug3("readPlatform() Executing '" + q + "'");
-			try (ResultSet rs = stmt.executeQuery(q);)
-			{
-
-				// Can only be 1 platform with a given ID.
-				if (rs != null && rs.next())
-				{
-					p.setAgency(rs.getString(2));
+		String q = "SELECT " + getColTpl() + " FROM Platform WHERE ID = ?";
+		debug3("readPlatform() Executing '" + q + "'");
 
-					p.isProduction = TextUtil.str2boolean(rs.getString(3));
-
-					DbKey siteId = DbKey.createDbKey(rs, 4);
-					if (!rs.wasNull())
-					{
-						// If site was previously loaded, use it.
-						p.setSite(p.getDatabase().siteList.getSiteById(siteId));
+		getDaoHelper().doQuery(q,rs-> readPlatform(p,rs),p.getId());
+		readPlatformSensors(p);
+	}
 
-						SiteDAI siteDAO = _dbio.makeSiteDAO();
-						siteDAO.setManualConnection(connection);
-						try
-						{
-							if (p.getSite() == null)
-							{
-								if (!DbKey.isNull(siteId))
-								{
-									try
-									{
-										p.setSite(siteDAO.getSiteById(siteId));
-										p.getDatabase().siteList.addSite(p.getSite());
-									}
-									catch(Exception ex)
-									{
-										warning("Platform " + p.getDisplayName() + " id="
-											+ p.getKey() + " has invalid siteID " + siteId);
-										p.setSite(null);
-									}
-								}
-							}
-							else
-							{
-								try
-								{
-									siteDAO.readSite(p.getSite());
-								}
-								catch (Exception e)
-								{
-									// TODO Auto-generated catch block
-									e.printStackTrace();
-								}
-							}
-						}
-						finally
-						{
-							siteDAO.close();
+	private void readPlatform(Platform p, ResultSet rs) throws SQLException, DatabaseException
+	{
+		// Can only be 1 platform with a given ID.
+		p.setAgency(rs.getString(2));
+
+		p.isProduction = TextUtil.str2boolean(rs.getString(3));
+
+		DbKey siteId = DbKey.createDbKey(rs, 4);
+		if (!rs.wasNull()) {
+			// If site was previously loaded, use it.
+			p.setSite(p.getDatabase().siteList.getSiteById(siteId));
+
+			SiteDAI siteDAO = _dbio.makeSiteDAO();
+			siteDAO.setManualConnection(connection);
+			try {
+				if (p.getSite() == null) {
+					if (!DbKey.isNull(siteId)) {
+						try {
+							p.setSite(siteDAO.getSiteById(siteId));
+							p.getDatabase().siteList.addSite(p.getSite());
+						} catch (Exception ex) {
+							warning("Platform " + p.getDisplayName() + " id="
+									+ p.getKey() + " has invalid siteID " + siteId);
+							p.setSite(null);
 						}
 					}
+				} else {
+					try {
+						siteDAO.readSite(p.getSite());
+					} catch (Exception e) {
+						// TODO Auto-generated catch block
+						e.printStackTrace();
```

Comment:
This should log to the logger, not be printed to stdout.

Also, for this block, bracket placement.